### PR TITLE
randomly pick up port number for dynamic registry

### PR DESCRIPTION
**Pull request by stevehu**

### CODE REVIEW

File: server/src/main/java/com/networknt/server/Server.java [hunk 1]
```diff
@@ -262,8 +269,8 @@ static private boolean bind(HttpHandler handler, int port) {
             server.start();
             System.out.println("HOST IP " + System.getenv(STATUS_HOST_IP));
         } catch (Exception e) {
-            if (!serverConfig.dynamicPort || (serverConfig.dynamicPort && serverConfig.maxPort == port + 1)) {
-                String triedPortsMessage = serverConfig.dynamicPort ? serverConfig.minPort + " to: " + (serverConfig.maxPort - 1) : port + "";
+            if (!serverConfig.dynamicPort || bindCounter > (serverConfig.maxPort - serverConfig.maxPort)) {
```
Comment (jiachen1120) [inline server/src/main/java/com/networknt/server/Server.java]: Hi! Is this a typo? I think it should be `serverConfig.maxPort - serverConfig.minPort`

Comment (stevehu) [reply server/src/main/java/com/networknt/server/Server.java]: It was my mistake. Thanks a lot for pointing it out.

**Comment by stevehu** [comment]
With this PR, the behavior changed a little bit. Previously, the max port is never been used and now it is been used. Please let me know if you guys see any negative impact. Thanks.

File: server/src/main/java/com/networknt/server/Server.java [hunk 2]
```diff
@@ -173,8 +175,13 @@ static public void start() {
...
                 throw new RuntimeException(errMessage);
             }          
-            for (int i = serverConfig.minPort; i < serverConfig.maxPort; i++) {
-                boolean b = bind(gracefulShutdownHandler, i);
+            while(true) {
+                bindCounter++;
+                if(bindCounter > (serverConfig.maxPort - serverConfig.minPort)) {
+                    break;
+                }
+                int randomPort = ThreadLocalRandom.current().nextInt(serverConfig.minPort, serverConfig.maxPort + 1);
```
Comment (jiachen1120) [inline server/src/main/java/com/networknt/server/Server.java]: After research on TreadLoadRandom, if serverConfig.minPort >= serverConfig.maxPort + 1, an IllegalArgumentException would be throw. Should we catch and handle this exception?

Comment (stevehu) [reply server/src/main/java/com/networknt/server/Server.java]: I think it is a good idea. We shouldn't allow the user to start the server in this case. I will make a change and test it to ensure the server stops. Thanks.

Comment (stevehu) [reply server/src/main/java/com/networknt/server/Server.java]: There is a check before that. 

```
            if (serverConfig.minPort > serverConfig.maxPort) {
                String errMessage = "No ports available to bind to - the minPort is larger than the maxPort in server.yml";
                System.out.println(errMessage);
                logger.error(errMessage);
                throw new RuntimeException(errMessage);
            }          

```

Comment (jiachen1120) [reply server/src/main/java/com/networknt/server/Server.java]: Oh, sorry, I didn't notice that

**Pull request merged**